src/components/catalog-component.js

```diff
@@ -6,28 +6,51 @@ import withCatalog from './with-catalog'
  *
  * - rendering requested component from the catalog
  * - passing props to the requested component
+ * - render a fallback component when `props.fallback` (node) is defined
  * - returns null (does not throw an error) when component does not exist in
  *   the catalog
  *
  * Example:
  *
- * <CatalogComponent component="Button" hello="world" />
+ * <CatalogComponent component="Button" hello="world" fallback={() => <div>not found</div>} />
  *
  * CatalogComponent will now render the `<Button />` component (if it is part of
- * the catalog and pass `hello` as a prop to the Button).
+ * the catalog and pass `hello` as a prop to the Button). If the component would
+ * not exist it will render the fallback.
  */
-class CatalogComponent extends React.Component {
+export class CatalogComponent extends React.Component {
   render() {
-    const { catalog, component, ...others } = this.props
+    const {
+      // catalog props
+      catalog,
+      component,
+      fallbackComponent: FallbackComponent,
+      // other props passed to component
+      ...others
+    } = this.props
 
-    if (catalog && catalog.getComponent) {
-      const Component = catalog.getComponent(component)
+    if (!catalog || !catalog.getComponent) {
+      console.error(
+        'catalog is not defined. Please use <CatalogComponent /> in the context of a <CatalogProvider /> with an existing catalog.',
+      )
+      return null
+    }
+
+    const Component = catalog.getComponent(component)
+    if (Component) {
+      return <Component {...others} />
+    }
 
-      if (Component) {
-        return <Component {...others} />
-      }
+    if (FallbackComponent) {
+      return <FallbackComponent {...others} />
     }
 
+    // if no component was found, tell the developer and fail gracefully
+    console.warn(
```

Comment:
As a future feature, how about using https://www.npmjs.com/package/debug here? Users could then control the log output. also, not sure if it is a good idea to output all available components 🤔 granted it can be quite useful but also spam the log files quickly if the catalog is big and the component call for a not available component is done often. especially if a use can not control the log behaviour

Reply:
Ack, will consider it in the future.